src/utils/auth.js

```diff
@@ -1,8 +1,45 @@
+const { SECRET } = require('../config');
+const { AuthorizationError } = require('./errors');
+const jwt = require('jsonwebtoken');
+const { Logger } = require("../utils/log4js.js");
+const log = Logger();
+
 // middleware that checks required authentication
 function requireAuth(req, res, next) {
-  //Invalid auth response example:
-  //return res.status(401).json({ error: 'Unauthorized request' });
-  next();
+  try {
+    // verify token exists
+    const token = req.headers['authorization']?.replace('Bearer ', '')
+    if (token === undefined) {
+      log.debug('JWT not provided');
+      throw new AuthorizationError;
+    }
+
+    // verify token is valid
+    jwt.verify(token, SECRET, function (err, decoded) {
+      if (err) {
+        log.debug('JWT verification error');
```

Comment:
It's super small thing but we can use log.error here. log4js has different log levels: https://www.tutorialspoint.com/log4j/log4j_logging_levels.htm